### ISSUE TITLE
s.remove_background non-interactive: lose metadata and axes_manager info

### ISSUE THREAD

**Issue by magnunor** [hyperspy/hyperspy#2104]
The output signal from `s.remove_background(signal_range=(..., ...))` does not retain either the metadata or axes_manager info.

Does occurs both in `RELEASE_next_patch`, the most recent HyperSpy bundle, and most recent Anaconda release.

The immediate consequence is that the EELS notebook is currently broken for the workshop next week.

```python
import numpy as np
import hyperspy.api as hs

s = hs.signals.Signal1D(np.random.random(size=(10, 500)))
s.axes_manager[0].name = 'axis0'
s.axes_manager[1].name = 'axis1'
s.axes_manager[0].units = 'units0'
s.axes_manager[1].units = 'units1'
s.axes_manager[0].scale = 0.9
s.axes_manager[1].scale = 1.1
s.metadata.General.title = "atitle"

s_r = s.remove_background(signal_range=(1, 100))
```

`s.axes_manager` output:
```
<Axes manager, axes: (10|500)>
            Name |   size |  index |  offset |   scale |  units
================ | ====== | ====== | ======= | ======= | ======
           axis0 |     10 |      0 |       0 |     0.9 | units0
---------------- | ------ | ------ | ------- | ------- | ------
           axis1 |    500 |        |       0 |     1.1 | units1
```

`s_r.axes_manager` output:
```
<Axes manager, axes: (10|500)>
            Name |   size |  index |  offset |   scale |  units
================ | ====== | ====== | ======= | ======= | ======
     <undefined> |     10 |      0 |       0 |       1 | <undefined>
---------------- | ------ | ------ | ------- | ------- | ------
     <undefined> |    500 |        |       0 |       1 | <undefined>
```

`s.metadata.General.title` output:
```
'atitle'
```

`s_r.metadata.General.title` output:
```
''
```

**Comment by thomasaarholt** [comment]
Any idea on how this came about? This can't have been the case all along?

**Comment by francisco-dlp** [comment]
It's amazing how buggy such a simple function can be. I rarely use this functionality but today I've used it and I found that with fast enabled the background was not properly subtracted, so one more bug.

**Comment by ericpre** [comment]
Fixed in #2156.
